src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,371 @@
+                    this.nextStep();
+                }
+            });
+        } else {
+            if (this.currentTour.completeCallback) {
+                this.currentTour.completeCallback();
+            }
+            this.updateGuidedTourSettings(this.currentTour.settingsId, false).subscribe(guidedTourSettings => {
+                if (guidedTourSettings.body) {
+                    this.guidedTourSettings = guidedTourSettings.body;
+                }
+            });
+            this.resetTour();
+        }
+    }
+
+    /**
+     * Navigate to previous tour step
+     */
+    public backStep(): void {
+        if (this.currentTour) {
+            const currentStep = this.currentTour.steps[this.currentTourStepIndex];
+            if (currentStep.closeAction) {
+                currentStep.closeAction();
+            }
+            if (this.currentTour.steps[this.currentTourStepIndex - 1]) {
+                this.currentTourStepIndex--;
+                if (currentStep.action) {
+                    currentStep.action();
+                }
+                setTimeout(() => {
+                    if (this.checkSelectorValidity()) {
+                        this.guidedTourCurrentStepSubject.next(this.getPreparedTourStep(this.currentTourStepIndex));
+                    } else {
+                        this.backStep();
+                    }
+                });
+            } else {
+                this.resetTour();
+            }
+        }
+    }
+
+    /**
+     * Skip tour and exit the tour context
+     */
+    public skipTour(): void {
+        if (this.currentTour) {
+            if (this.currentTour.skipCallback) {
+                this.currentTour.skipCallback(this.currentTourStepIndex);
+            }
+            this.resetTour();
+        }
+    }
+
+    /**
+     * Close tour by resetting `currentTour`, `currentTourStepIndex` and `guidedTourCurrentStepSubject`
+     * and remove overlay
+     */
+    public resetTour(): void {
+        document.body.classList.remove('tour-open');
+        this.currentTour = null;
+        this.currentTourStepIndex = 0;
+        this.guidedTourCurrentStepSubject.next(null);
+    }
+
+    /**
+     * Start guided tour for given guided tour
+     * @param {tour} guided tour
+     */
+    private startTour(tour: GuidedTour): void {
+        this.currentTourSteps = tour.steps;
+
+        // adjust tour steps according to permissions
+        tour.steps.forEach((step, index) => {
+            if (step.permission && !this.accountService.hasAnyAuthorityDirect(step.permission)) {
+                this.currentTourSteps.splice(index, 1);
+            }
+        });
+
+        this.currentTour = cloneDeep(tour);
+        this.currentTour.steps = this.currentTour.steps.filter(step => !step.skipStep);
+        this.currentTourStepIndex = 0;
+        if (this.currentTour.steps.length > 0 && (!this.currentTour.minimumScreenSize || window.innerWidth >= this.currentTour.minimumScreenSize)) {
```

Comment:
This if statement either needs a comment or should better be hidden in a function with an explanatory name.

Reply:
@thilo-behnke I changed that